File: src/main/java/gg/uhc/migration/selection/GlobalAreaSelection.java
package gg.uhc.migration.selection;

import gg.uhc.migration.Area;

import java.util.UUID;

public class GlobalAreaSelection extends AreaSelection {

    protected Area chosenArea = null;

    @Override
    public Area getForUUID(UUID uuid) {
        if (chosenArea == null) {
            randomize();
        }

        return chosenArea;
    }

    @Override
    public void randomize() {
        chosenArea = this.potentialAreas.random();
    }
}

File: src/main/java/gg/uhc/migration/selection/GlobalAreaSelection.kt
package gg.uhc.migration.selection

import gg.uhc.migration.Area

import java.util.UUID

class GlobalAreaSelection : AreaSelection() {
    private var chosenArea: Area? = null

    override fun getForUUID(uuid: UUID): Area {
        if (chosenArea == null) {
            randomize()
        }

        return chosenArea!!
    }

    override fun randomize() {
        chosenArea = this.potentialAreas.get()
    }
}
